File: dotnet-desktop-guide/net/wpf/events/snippets/weak-event-patterns/csharp/SomeEventSource.cs
using System;

namespace CodeSampleCsharp
{
    public class SomeEventSource
    {
        public event EventHandler<SomeEventArgs> SomeEvent;

        public void RaiseDoEvent()
        {
            OnSomeEvent(new SomeEventArgs());
        }

        protected void OnSomeEvent(SomeEventArgs e)
        {
            SomeEvent?.Invoke(this, e);
        }
    }
}

File: dotnet-desktop-guide/net/wpf/events/snippets/weak-event-patterns/vb/SomeEventSource.vb
Public Class SomeEventSource
    Public Event SomeEvent As EventHandler(Of SomeEventArgs)

    Public Sub RaiseDoEvent()
        OnSomeEvent(New SomeEventArgs())
    End Sub

    Protected Sub OnSomeEvent(e As SomeEventArgs)
        RaiseEvent SomeEvent(Me, e)
    End Sub
End Class
